### ISSUE TITLE
SelectViewHelper is missing the required attribute

### ISSUE THREAD

**Issue by knusperpixel** [neos/flow-development-collection#842]
### Description

When using the SelectViewHelper it is not possible to add the required attribute.

### Steps to Reproduce

1. Add the view helper to your template `<f:form.select property="someProperty" options="{someOptions}" required="true" />`

#### Expected behavior

The html output should contain the `required` attribute

#### Actual behavior

Exception: `The argument "required" was not registered`

### Affected Versions

Neos: 2.2.5

Flow: 3.2.5

**Comment by kdambekalns** [comment]
None of the VHs support that, except through `additionalAttributes` (unless I cannot mentally parse the code anymore). That being said, it would be great to add that for all input VHs, except `hidden`, `range`, `color` types and the `<button>` elements (see https://html.spec.whatwg.org/multipage/forms.html#concept-input-apply)

**Comment by knusperpixel** [comment]
Well, at least the TextfieldViewHelper does support the `required` attribute:
https://github.com/neos/flow-development-collection/blob/2.0/TYPO3.Fluid/Classes/TYPO3/Fluid/ViewHelpers/Form/TextfieldViewHelper.php#L78

**Comment by kdambekalns** [comment]
True. It was my mental parsing that failed, I just checked the registered arguments, not the parameters to `render()`…

**Comment by kdambekalns** [comment]
Now, the interesting question is this: how does this behave for Flow 4.0 with the use of `typo3/typo3fluid`? @kitsunet how much abstraction do we need to adjust to get "all features" of the "upstream library"?

**Comment by kitsunet** [comment]
As form is tightly coupled to MVC it's fully our responsibility. There are zero form related VHs in typo3fluid. So we just implement it in our adaptor and are done.

**Comment by albe** [comment]
One back question regarding implementation: in TextfieldViewHelper it's currently a *boolean* `render()` method argument. Should we keep it like that for all VHs or change to use `registerTagAttribute`? And if the latter, should we use `boolean` or `string` type - the latter is also used for the `disabled`, so that would be mis matching.

**Comment by kitsunet** [comment]
`registerTagAttribute` is the now preferred way even if that is not communicated yet. Logic for render method arguments is ours.  In general no render method in favor of static render is preferred. 

As for string/boolean, no clue, depends on how it's used I guess. The only valid value for disabled is "disabled" right? so IMHO a bool should be enough?

**Comment by albe** [comment]
Well, according to HTML5/Whatwg the `required` attribute (like `disabled`) is considered a "boolean attribute" (https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#boolean-attribute), which denotes a true value by the sole presence. It is allowed to provide it without a value or with an empty string value or the case-insensitive match of the attribute name. Typo3Fluid currently always renders tag attributes with values.

Should we change the "disabled" attribute then too?

**Comment by Benjamin-K** [comment]
Also was looking for this. IMHO this should be implemented for all form elements that support `required` via `registerTagAttribute` (as boolean). The render method for those that had `$required` as an argument were already updated. Now we need to add the `required` attribute (and maybe `disabled`, too) to the form elements that still miss this attribute.

After this is done, the templates of the [Flow Form Framework](https://github.com/neos/form/tree/master/Resources/Private/Form) should be updated to add the required attributes directly to the supported elements. (Maybe this behaviour can be adjusted through a setting as some developers do not want to use client side validation as the layout depends on the used browser.)
